packages/composites/glv-token/src/transport/price.ts

```diff
@@ -0,0 +1,374 @@
+    const minimizedPrice = Number(utils.formatUnits(minimizedValue, SIGNED_PRICE_DECIMALS))
+    const result = median([minimizedPrice, maximizedPrice])
+
+    return {
+      data: {
+        result: result,
+        sources: priceResult.sources,
+      },
+      statusCode: 200,
+      result: result,
+      timestamps: {
+        providerDataRequestedUnixMs,
+        providerDataReceivedUnixMs: Date.now(),
+        providerIndicatedTimeUnixMs: undefined,
+      },
+    }
+  }
+
+  // Fetches the lwba price info from multiple source EAs, calculates the median for bids and asks per asset and fixes the price precision
+  private async fetchPrices(assets: string[], dataRequestedTimestamp: number) {
+    // priceData holds raw bid/ask values per asset from source EAs response
+    const priceData = {} as PriceData
+
+    const sources = [
+      { url: this.settings.TIINGO_ADAPTER_URL, name: 'tiingo' },
+      { url: this.settings.COINMETRICS_ADAPTER_URL, name: 'coinmetrics' },
+      { url: this.settings.NCFX_ADAPTER_URL, name: 'ncfx' },
+    ]
+
+    //priceProviders contains assets with a list of sources where asset price was successfully fetched
+    const priceProviders: Record<string, string[]> = {}
+
+    const promises = []
+
+    for (let i = 0; i < sources.length; i++) {
+      const source = sources[i]
+
+      const assetPromises = assets.map(async (asset) => {
+        const base = this.unwrapAsset(asset)
+        const requestConfig = {
+          url: source.url,
+          method: 'POST',
+          data: {
+            data: {
+              endpoint: 'crypto-lwba',
+              base,
+              quote: 'USD',
+            },
+          },
+        }
+
+        // try/catch is needed in a case if one of source EAs fails to return a response,
+        // we will still get and calculate the median price based on responses of remaining EAs (based on MIN_REQUIRED_SOURCE_SUCCESS setting)
+        try {
+          const response = await this.requester.request<{ data: LwbaResponseDataFields['Data'] }>(
+            JSON.stringify(requestConfig),
+            requestConfig,
+          )
+          const { bid, ask } = response.response.data.data
+
+          priceData[asset] = {
+            bids: [...(priceData[asset]?.bids || []), bid],
+            asks: [...(priceData[asset]?.asks || []), ask],
+          }
+
+          priceProviders[base] = priceProviders[base]
+            ? [...new Set([...priceProviders[base], source.name])]
+            : [source.name]
+        } catch (error) {
+          const e = error as Error
+          logger.error(
+            `Error fetching data for ${asset} from ${source.name}, url - ${source.url}: ${e.message}`,
+          )
+        }
+      })
+
+      promises.push(...assetPromises)
+    }
+
+    await Promise.all(promises)
+
+    this.validateRequiredResponses(priceProviders, sources, assets, dataRequestedTimestamp)
+
+    const medianValues = this.calculateMedian(assets, priceData)
```

Comment:
Can we re-use the Medianizer Composite Adapter?

Reply:
I would actually avoid using one more composite adapter as it overcomplicates the setup. Also the Medianizer is a v2 adapter and we should avoid using v2 EAs if possible